packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts

```diff
@@ -115,17 +115,27 @@ export default async (req, res) => {
       existingFulfillments.map((fulfillment) => [fulfillment.id, fulfillment])
     )
 
-    const { fulfillments } = await orderService
+    await orderService
       .withTransaction(transactionManager)
       .createFulfillment(id, validated.items, {
         metadata: validated.metadata,
         no_notification: validated.no_notification,
       })
 
-    const pvInventoryServiceTx =
-      pvInventoryService.withTransaction(transactionManager)
-
     if (validated.location_id) {
+      const { fulfillments } = await orderService
```

Comment:
todo: I can see that we create multiple instance of that service, would it be better to create a const of it?